### PR TITLE
RBAC: Route `Claims` into `TableOfContent` for points API

### PR DESCRIPTION
**Pull request by coszio**
Tracked by #3777 

There are some claims, like `filter` and `payload` which need to access the deserialized structures from `Filter` and `Payload`. To avoid code duplication in both tonic and actix layers, we'll handle these cases in the table of contents.

- Extracts claims from the extensions of the request in tonic and actix interfaces
- Routes them into table of contents

### All Submissions:

* [x] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/actix/api/count_api.rs [hunk 1]
```diff
@@ -35,7 +35,8 @@ async fn count_points(
         count_request,
         params.consistency,
         shard_selector,
-        // ToDo: use timeout from params
+        None, // TODO(RBAC): add claims from request
+              // ToDo: use timeout from params
```
Comment (xzfc) [inline src/actix/api/count_api.rs:39]: Style nit: don't move this comment.

File: src/tonic/api/points_internal_api.rs [hunk 2]
```diff
@@ -37,9 +38,12 @@ impl PointsInternalService {
 impl PointsInternal for PointsInternalService {
     async fn upsert(
         &self,
-        request: Request<UpsertPointsInternal>,
+        mut request: Request<UpsertPointsInternal>,
     ) -> Result<Response<PointsOperationResponseInternal>, Status> {
         validate_and_log(request.get_ref());
+
+        let claims = request.extensions_mut().remove::<Claims>();
```
Comment (xzfc) [inline src/tonic/api/points_internal_api.rs:45]: I suggest to put code that calls `extensions_mut().insert()` and `extensions_mut().remove()` to the same module. Then this line should look like this:

```suggestion
        let claims = super::auth::extract_claims(&mut request);
```

**Review by xzfc — CHANGES_REQUESTED** [review]

**Review by xzfc — APPROVED** [review]

**Comment by coszio** [comment]
Thanks for the rebase @xzfc 🙌, I'll just do a quick refactor of `ActixClaims` into `Extension<T>` in case we want to reuse it later

File: src/tonic/api/points_internal_api.rs [hunk 3]
```diff
@@ -21,6 +21,7 @@ use crate::tonic::api::points_common::{
     delete_payload, delete_vectors, get, overwrite_payload, recommend, scroll, set_payload, sync,
     update_vectors, upsert,
 };
+use crate::tonic::auth::extract_claims;
```
Comment (generall) [inline src/tonic/api/points_internal_api.rs:24]: I don't think we need Claim checks for internal API

**Review by generall — APPROVED** [review]

**Pull request merged**